### PR TITLE
 Driver for the SRF02 ultrasound range finder, adapted from Maxbotix driver

### PR DESCRIPTION
**Pull request by gcarmix**
I created this driver for the SRF02 ultrasound range finder, I took the Maxbotix driver (mb12xx) and modified the I2C commands to make it compatible with the SRF02. I tried it with one sensor and it works.

Note: 
I also noticed that the original driver has a problem with the "test" command, since it always return 0.0 meters when you connect just one sensor. I corrected it on my driver, it should be corrected even in the mb12xx driver but I can't, because then I don't have a mb12xx sensor to test it

### CODE REVIEW

File: src/drivers/srf02/module.mk [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+
+#
+# Makefile to build the Maxbotix Sonar driver.
+#
+
+MODULE_COMMAND	= srf02
```
Comment (LorenzMeier) [inline src/drivers/srf02/module.mk]: We do not use the old build system any more. You need to use a CMakeLists.txt file.

**Comment by LorenzMeier** [comment]
Thanks!

**Comment by gcarmix** [comment]
You're welcome :)

File: src/drivers/srf02/srf02.cpp [hunk 2]
```diff
@@ -0,0 +1,961 @@
...
+	int ret;
+
+	/*
+	 * Send the command to begin a measurement.
+	 */
+
+	uint8_t cmd[2];
+	cmd[0] = 0x00;
+	cmd[1] = SRF02_TAKE_RANGE_REG;
+	ret = transfer(cmd, 2, nullptr, 0);
```
Comment (bansiesta) [inline src/drivers/srf02/srf02.cpp]: I'd move int from line 537 down here.

File: src/drivers/srf02/srf02.cpp [hunk 3]
```diff
@@ -0,0 +1,961 @@
...
+
+	if (OK != ret) {
+		perf_count(_comms_errors);
+		DEVICE_DEBUG("i2c::transfer returned %d", ret);
+		return ret;
+	}
+
+	ret = OK;
```
Comment (bansiesta) [inline src/drivers/srf02/srf02.cpp]: ret is already OK here

File: src/drivers/srf02/srf02.cpp [hunk 4]
```diff
@@ -0,0 +1,961 @@
...
+	/* do a simple demand read */
+	sz = read(fd, &report, sizeof(report));
+
+	if (sz != sizeof(report)) {
+		err(1, "immediate read failed");
+	}
+
+	warnx("single read");
+	warnx("measurement: %0.2f m", (double)report.current_distance);
+	warnx("time:        %llu", report.timestamp);
```
Comment (bansiesta) [inline src/drivers/srf02/srf02.cpp]: Please add "us" for unit.

File: src/drivers/srf02/srf02.cpp [hunk 5]
```diff
@@ -0,0 +1,961 @@
...
+		sz = read(fd, &report, sizeof(report));
+
+		if (sz != sizeof(report)) {
+			err(1, "periodic read failed");
+		}
+
+		warnx("periodic read %u", i);
+		warnx("valid %u", (float)report.current_distance > report.min_distance
+		      && (float)report.current_distance < report.max_distance ? 1 : 0);
+		warnx("measurement: %0.3f", (double)report.current_distance);
```
Comment (bansiesta) [inline src/drivers/srf02/srf02.cpp]: Please add "m" for unit.

File: src/drivers/srf02/srf02.cpp [hunk 6]
```diff
@@ -0,0 +1,961 @@
...
+		/* now go get it */
+		sz = read(fd, &report, sizeof(report));
+
+		if (sz != sizeof(report)) {
+			err(1, "periodic read failed");
+		}
+
+		warnx("periodic read %u", i);
+		warnx("valid %u", (float)report.current_distance > report.min_distance
+		      && (float)report.current_distance < report.max_distance ? 1 : 0);
```
Comment (bansiesta) [inline src/drivers/srf02/srf02.cpp]: You could also use "valid %s" and then .... ? "YES" : "NO"

**Comment by bansiesta** [comment]
I had a read through this and made some really minor comments. Looks good to me!

**Pull request merged**